eqwalizer/src/main/scala/com/whatsapp/eqwalizer/ast/ConvertAst.scala

```diff
@@ -607,6 +610,10 @@ class ConvertAst(fromBeam: Boolean, noAutoImport: Set[Id] = Set.empty) {
         LGenerate(convertPat(ePat), convertExp(eExp))
       case ETuple(List(EAtom("b_generate"), EPos(p), ePat, eExp)) =>
         BGenerate(convertPat(ePat), convertExp(eExp))
+      case ETuple(
+            List(EAtom("m_generate"), EPos(p), ETuple(List(EAtom("map_field_exact"), EPos(_), kPat, vPat)), eExp)
+          ) =>
+        MGenerate(convertPat(kPat), convertPat(vPat), convertExp(eExp))
```

Comment:
I'm not sure this is correct. Shouldn't key be an expression, like for regular map pattern matching?

Reply:
As we discussed - it should be a "test expression"

Reply:
Ok, - it turns out that it is "just a pattern" actually:

```erlang
-module(maps_examples).
-compile([export_all, nowarn_export_all]).

f1(#{(#{})#{a => b} := V}) -> V.

f2(M) -> #{ 1 => V || ((#{})#{a => b}) := V <- M }.
```

```
maps_examples.erl:6:29: illegal pattern
%    6| f2(M) -> #{ 1 => V || ((#{})#{a => b}) := V <- M }.
%     |                             ^

```

Reply:
Indeed looks like some inconsistency in the compiler itself. Let's leave it as pattern for now in eqwalizer and follow up with the OTP team

Reply:
Map comprehension generators are sugar for maps:to_list, see @bjorng comment after I asked something similar: https://github.com/erlang/otp/pull/6727#issuecomment-1420243389. So I think it makes sense that this is a pattern.

Reply:
Ah, I see. It is somewhat surprising, but I guess it makes sense given the implementation